From c3e982ebb4990825bbedd83e8741162d249fcfa2 Mon Sep 17 00:00:00 2001
From: Vishwath Mohan <vishwath@google.com>
Date: Thu, 2 Nov 2017 07:17:38 -0700
Subject: [PATCH 2/4] Fix TASKPTR's definition to match actual function
 signatures.

TASKPTR is currently defined as returning void. However the only 2
functions that are ever cast to this type return uint32_t, so this is
technically incorrect. This CL changes TASKPTR to match.

For P, we plan to enable control-flow integrity (CFI) across multiple
platform components, including NFC. This mismatch will break CFI and
cause runtime errors when enabled - so this is also a preemptive
measure to get NFC compatible in preparation.

Bug: 30227045
Test: Builds and runs without CFI errors when enabled.
Change-Id: I3111b5810937c923fdf8cbbd802520fad9baf272
---
 src/gki/common/gki.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gki/common/gki.h b/src/gki/common/gki.h
index 1458e32..4a4763a 100644
--- a/src/gki/common/gki.h
+++ b/src/gki/common/gki.h
@@ -337,7 +337,7 @@ typedef struct {
 /* Task constants
 */
 #ifndef TASKPTR
-typedef void (*TASKPTR)(uint32_t);
+typedef uint32_t (*TASKPTR)(uint32_t);
 #endif
 
 /* General pool accessible to GKI_getbuf() */
-- 
2.17.0

